src/market/directory.rs

```diff
@@ -25,7 +25,9 @@ use crate::utill::{
 /// Represents errors that can occur during directory server operations.
 #[derive(Debug)]
 pub enum DirectoryServerError {
-    Other(&'static str),
+    MutexPossion,
+    Socks5ConnectionError(String),
+    IO(std::io::Error),
```

Comment:
Refer to comments here: https://github.com/citadel-tech/coinswap/pull/241#pullrequestreview-2274395460

We should add the NetError instead of IO. Net already includes IO. 

Also better to merge this work with #241 as they are interdependent.